var/spack/repos/builtin/packages/pism/package.py

```diff
@@ -0,0 +1,175 @@
+    # (except from CMakeLists.txt)
+    #
+    # option (Pism_TEST_USING_VALGRIND "Add extra regression tests
+    #         using valgrind" OFF)
+    # mark_as_advanced (Pism_TEST_USING_VALGRIND)
+    #
+    # option (Pism_ADD_FPIC "Add -fPIC to C++ compiler flags
+    #         (CMAKE_CXX_FLAGS). Try turning it off if it does not work." ON)
+    # option (Pism_LINK_STATICALLY
+    #         "Set CMake flags to try to ensure that everything is
+    #         linked statically")
+    # option (Pism_LOOK_FOR_LIBRARIES
+    #         "Specifies whether PISM should look for libraries. (Disable
+    #         this on Crays.)" ON)
+    # option (Pism_USE_TR1
+    #        "Use the std::tr1 namespace to access shared pointer
+    #        definitions. Disable to get shared pointers from the std
+    #        namespace (might be needed with some compilers)." ON)
+    # option (Pism_USE_TAO "Use TAO in inverse solvers." OFF)
+
+    depends_on('fftw')
+    depends_on('gsl')
+    depends_on('mpi')
+    depends_on('netcdf')    # Only the C interface is used, no netcdf-cxx4
+    depends_on('petsc')
+    depends_on('udunits2')
+    depends_on('proj')
+    depends_on('everytrace', when='+everytrace')
+
+    extends('python', when='+python')
+    depends_on('python@2.7', when='+python')
+    depends_on('py-matplotlib', when='+python')
+    depends_on('py-numpy', when='+python')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        return [
+            '-DCMAKE_C_COMPILER=%s' % spec['mpi'].mpicc,
+            '-DCMAKE_CXX_COMPILER=%s' % spec['mpi'].mpicxx,
+            # Fortran not needed for PISM...
+            # '-DCMAKE_Fortran_COMPILER=%s' % spec['mpi'].mpifc,
+            '-DPism_BUILD_EXTRA_EXECS=%s' %
+            ('YES' if '+extra' in spec else 'NO'),
+            '-DBUILD_SHARED_LIBS=%s' %
+            ('YES' if '+shared' in spec else 'NO'),
+            '-DPism_BUILD_PYTHON_BINDINGS=%s' %
+            ('YES' if '+python' in spec else 'NO'),
+            '-DPism_BUILD_ICEBIN=%s' %
+            ('YES' if '+icebin' in spec else 'NO'),
+            '-DPism_USE_PROJ4=%s' %
+            ('YES' if '+proj' in spec else 'NO'),
+            '-DPism_USE_PARALLEL_NETCDF4=%s' %
+            ('YES' if '+parallel-netcdf4' in spec else 'NO'),
+            '-DPism_USE_PNETCDF=%s' %
+            ('YES' if '+parallel-netcdf3' in spec else 'NO'),
+            '-DPism_USE_PARALLEL_HDF5=%s' %
+            ('YES' if '+parallel-hdf5' in spec else 'NO'),
+            '-DPism_BUILD_PDFS=%s' %
+            ('YES' if '+doc' in spec else 'NO'),
+            '-DPism_INSTALL_EXAMPLES=%s' %
+            ('YES' if '+examples' in spec else 'NO'),
+            '-DPism_USE_EVERYTRACE=%s' %
+            ('YES' if '+everytrace' in spec else 'NO')]
+
+    def setup_environment(self, spack_env, env):
+        """Add <prefix>/bin to the module; this is not the default if we
+        extend python."""
+        env.prepend_path('PATH', self.prefix.bin)
```

Comment:
This is not needed, and I don't think the docstring is true. If it is, please file a bug.

Reply:
I'm willing to take it out, maybe it was needed on some ancient version of Spack.